src/nodes/helper.ts

```diff
@@ -98,13 +92,14 @@ function extendEvent(event: IKalenderEvent, config: IcalEventsConfig, kalenderEv
 }
 
 export async function getICal(node: IcalNode, RED,n:any) {
-    const kalenderEvents = new KalenderEvents()
```

Comment:
Nice work, thanks a lot for digging into the issue 👍 
The only thing I'm not sure about is this. I'm not 100% sure if this doesn't cause anyside affects, when my lib kalender-events is a singleton here 😉 
But let's have a try